contracts/src/7683/T1ERC7683.sol

```diff
@@ -180,6 +224,103 @@ contract T1ERC7683 is BasicSwap7683, OwnableUpgradeable, PausableUpgradeable {
+            abi.decode(_innerMessage, (bool, bytes32[], bytes[]));
+
+        for (uint256 i = 0; i < _orderIds.length; i++) {
+            if (_settle) {
+                _handleSettleOrder(_originDomain, _sender, _orderIds[i], abi.decode(_ordersFillerData[i], (bytes32)));
+            }
+        }
+    }
+
+    /// @notice Retrieves the local domain identifier.
+    /// @dev This function overrides the `_localDomain` function from the parent contract.
+    /// @return The local domain ID.
+    function _localDomain() internal view override returns (uint32) {
+        return localDomain;
+    }
+
+    /**
+     * @notice Refunds a batch of expired GaslessCrossChainOrders on the chain where the orders were opened.
+     * The refunded status should not be changed here but rather on the origin chain. To allow the user to retry in
+     * case some error occurs.
+     * Ensuring the order is eligible for refunding in the origin chain is the responsibility of the caller.
+     * @param _orders An array of GaslessCrossChainOrders to refund.
+     * @param _proofs Array of encoded proofs of read to verify orders are not settled
+     */
+    function refund(GaslessCrossChainOrder[] memory _orders, bytes[] calldata _proofs) external payable {
+        if (_orders.length != _proofs.length) revert LengthMismatch();
+
+        bytes32[] memory orderIds = new bytes32[](_orders.length);
+        for (uint256 i = 0; i < _orders.length; i += 1) {
+            bytes32 orderId = _getOrderId(_orders[i]);
+            _verifyOrderNotFilled(orderId, _proofs[i]);
+            orderIds[i] = orderId;
+        }
+
+        _refundOrders(OrderEncoder.decode(_orders[0].orderData).originDomain, orderIds);
+    }
+
+    /**
+     * @notice Refunds a batch of expired OnchainCrossChainOrder on the chain where the orders were opened.
+     * The refunded status should not be changed here but rather on the origin chain. To allow the user to retry in
+     * case some error occurs.
+     * Ensuring the order is eligible for refunding the origin chain is the responsibility of the caller.
+     * @param _orders An array of OnchainCrossChainOrders to refund.
+     * @param _proofs Array of encoded proofs of read to verify orders are not settled
+     */
+    function refund(OnchainCrossChainOrder[] memory _orders, bytes[] calldata _proofs) external payable {
+        if (_orders.length != _proofs.length) revert LengthMismatch();
+
+        bytes32[] memory orderIds = new bytes32[](_orders.length);
+        for (uint256 i = 0; i < _orders.length; i += 1) {
+            bytes32 orderId = _getOrderId(_orders[i]);
+            _verifyOrderNotFilled(orderId, _proofs[i]);
+            orderIds[i] = orderId;
+        }
+
+        _refundOrders(OrderEncoder.decode(_orders[0].orderData).originDomain, orderIds);
+    }
+
+    /**
+     * @dev Should be implemented by the messaging layer for dispatching a refunding instruction the remote domain
```

Comment:
This description feels wrong. We are not dispatching a refund instruction, we are executing the refund.